crypto/dkg.go

```diff
@@ -53,6 +58,33 @@ type DKGState interface {
 	ForceDisqualify(node int) error
 }
 
+// dkgFailureError is an error returned when a participant
+// detects a failure in the protocol and is not able to compute output keys.
+// Such a failure can be local and only depends on the participant's view of what
+// happened in the protocol. The error can only be returned using the End() function.
+type dkgFailureError struct {
+	err error
+}
+
+// dkgFailureErrorf constructs a new dkgFailureError
+func dkgFailureErrorf(msg string, args ...interface{}) error {
+	return &dkgFailureError{
+		err: fmt.Errorf(msg, args...),
+	}
+}
+
+func (e dkgFailureError) Error() string {
+	return e.err.Error()
```

Comment:
```suggestion
	return fmt.Sprintf("dkg failure: %s", e.err.Error())
```
include the context of this error level

Reply:
I think the errors built using `dkgFailureErrorf` are self explanatory. I think wrapping the error message in a context is useful when the error itself is returned by another function, for instance [here](https://github.com/onflow/flow-go/blob/tarak/5931-dkg-failure-bug/engine/consensus/dkg/reactor_engine.go#L278).

On a different note, I think I could embedded the error in the structure `dkgFailureError` and avoid defining `Error()`.

Reply:
I think it's useful to have an error type include some context for example for searching logs. 

If I want to search for `dkgFailureError` errors, I can search for the "dkg failure: "` string

Reply:
added in https://github.com/onflow/flow-go/pull/1693/commits/52236feb1ee7f0e17337c5129eb951b6b56e55d8 with missing tests